tests/Meziantou.Analyzer.Test/Rules/AvoidUsingRedundantElseAnalyzerTests.cs

```diff
@@ -0,0 +1,285 @@
+﻿using System.Threading.Tasks;
+using Meziantou.Analyzer.Rules;
+using TestHelper;
+using Xunit;
+
+namespace Meziantou.Analyzer.Test.Rules
+{
+    public sealed class AvoidUsingRedundantElseAnalyzerTests
+    {
+        private static ProjectBuilder CreateProjectBuilder()
+        {
+            return new ProjectBuilder()
+                .WithAnalyzer<AvoidUsingRedundantElseAnalyzer>();
+        }
+
+        [Fact]
+        public async Task IfBreakElse_DiagnosticIsReported()
+        {
+            var sourceCode = @"
+class TestClass
+{
+    void Test()
+    {
+        var value = -1;
+        while (true)
+        {
+            if (value < 0)
+            {
+                {
```

Comment:
I created an extra block level here to validate that the jump statement need not be directly in the `if` block to be detected.

Reply:
`IsDirectAccess` is only needed because of the local function. If you remove it you can see that `IsEndPointReachable` is false. This seems like a weird behavior of Roslyn. 

`IsDirectAccess` is fine with me.